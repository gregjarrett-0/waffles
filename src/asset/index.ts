export { default as BarChart3d } from './generated/bar-chart-3d';
export { default as Bookmark3d } from './generated/bookmark-3d';
export { default as Branch3d } from './generated/branch-3d';
export { default as Calendar3d } from './generated/calendar-3d';
export { default as Certification3d } from './generated/certification-3d';
export { default as Checkmark3d } from './generated/checkmark-3d';
export { default as Code3d } from './generated/code-3d';
export { default as Database3d } from './generated/database-3d';
export { default as Direction3d } from './generated/direction-3d';
export { default as DoubleArrow3d } from './generated/double-arrow-3d';
export { default as Edit3d } from './generated/edit-3d';
export { default as Education3d } from './generated/education-3d';
export { default as Email3d } from './generated/email-3d';
export { default as Finance3d } from './generated/finance-3d';
export { default as FourArrows3d } from './generated/four-arrows-3d';
export { default as Globe3d } from './generated/globe-3d';
export { default as Import3d } from './generated/import-3d';
export { default as Laptop3d } from './generated/laptop-3d';
export { default as Love3d } from './generated/love-3d';
export { default as MagnifyingGlass3d } from './generated/magnifying-glass-3d';
export { default as MailboxEmpty3d } from './generated/mailbox-empty-3d';
export { default as Mobile3d } from './generated/mobile-3d';
export { default as Paper3d } from './generated/paper-3d';
export { default as Path3d } from './generated/path-3d';
export { default as People3d } from './generated/people-3d';
export { default as Rocket3d } from './generated/rocket-3d';
export { default as Share3d } from './generated/share-3d';
export { default as TargetArrow3d } from './generated/target-arrow-3d';
export { default as Target3d } from './generated/target-3d';
export { default as Team3d } from './generated/team-3d';
export { default as TimerSleep3d } from './generated/timer-sleep-3d';
export { default as Timer3d } from './generated/timer-3d';
export { default as Track3d } from './generated/track-3d';
export { default as Trophy3d } from './generated/trophy-3d';
export { default as ApplyAlpa } from './generated/apply-alpa';
export { default as AssessAlpa } from './generated/assess-alpa';
export { default as LearnAlpa } from './generated/learn-alpa';
export { default as PracticeAlpa } from './generated/practice-alpa';
export { default as ApplyColoredAlpaLoop } from './generated/apply-colored-alpa-loop';
export { default as AssessColoredAlpaLoop } from './generated/assess-colored-alpa-loop';
export { default as LearnColoredAlpaLoop } from './generated/learn-colored-alpa-loop';
export { default as PracticeColoredAlpaLoop } from './generated/practice-colored-alpa-loop';
export { default as RegularColoredAlpaLoop } from './generated/regular-colored-alpa-loop';
export { default as AssessmentAssignment } from './generated/assessment-assignment';
export { default as ChapterAssignment } from './generated/chapter-assignment';
export { default as CourseAssignment } from './generated/course-assignment';
export { default as CustomTrackAssignment } from './generated/custom-track-assignment';
export { default as EarnXpAssignment } from './generated/earn-xp-assignment';
export { default as ProjectAssignment } from './generated/project-assignment';
export { default as TrackAssignment } from './generated/track-assignment';
export { default as BnpParibasLogo } from './generated/bnp-paribas-logo';
export { default as ColgatePalmoliveLogo } from './generated/colgate-palmolive-logo';
export { default as CreditSuisseLogo } from './generated/credit-suisse-logo';
export { default as DeloitteLogo } from './generated/deloitte-logo';
export { default as DockerLogo } from './generated/docker-logo';
export { default as EbayLogo } from './generated/ebay-logo';
export { default as ExcelLogo } from './generated/excel-logo';
export { default as GitLogo } from './generated/git-logo';
export { default as GoogleLogo } from './generated/google-logo';
export { default as HsbcLogo } from './generated/hsbc-logo';
export { default as JuliaLogo } from './generated/julia-logo';
export { default as MercedesBenzLogo } from './generated/mercedes-benz-logo';
export { default as MicrosoftLogo } from './generated/microsoft-logo';
export { default as OracleLogo } from './generated/oracle-logo';
export { default as PayPalLogo } from './generated/pay-pal-logo';
export { default as PowerBiLogo } from './generated/power-bi-logo';
export { default as PythonLogo } from './generated/python-logo';
export { default as RLogo } from './generated/r-logo';
export { default as ScalaLogo } from './generated/scala-logo';
export { default as ShellLogo } from './generated/shell-logo';
export { default as SparkLogo } from './generated/spark-logo';
export { default as SpreadsheetsLogo } from './generated/spreadsheets-logo';
export { default as SqlLogo } from './generated/sql-logo';
export { default as TMobileLogo } from './generated/t-mobile-logo';
export { default as TableauLogo } from './generated/tableau-logo';
export { default as UberLogo } from './generated/uber-logo';
export { default as WafflesColoredLogo } from './generated/waffles-colored-logo';
export { default as WafflesLogo } from './generated/waffles-logo';
export { default as DockerLogomark } from './generated/docker-logomark';
export { default as ExcelLogomark } from './generated/excel-logomark';
export { default as GitLogomark } from './generated/git-logomark';
export { default as JuliaLogomark } from './generated/julia-logomark';
export { default as PowerBiLogomark } from './generated/power-bi-logomark';
export { default as PythonLogomark } from './generated/python-logomark';
export { default as RLogomark } from './generated/r-logomark';
export { default as ScalaLogomark } from './generated/scala-logomark';
export { default as ShellLogomark } from './generated/shell-logomark';
export { default as SpreadsheetLogomark } from './generated/spreadsheet-logomark';
export { default as SqlLogomark } from './generated/sql-logomark';
export { default as TableauLogomark } from './generated/tableau-logomark';
export { default as TheoryLogomark } from './generated/theory-logomark';
export { default as WafflesColoredLogomark } from './generated/waffles-colored-logomark';
export { default as WafflesLogomark } from './generated/waffles-logomark';
export { default as AvatarPlaceholderColored } from './generated/avatar-placeholder-colored';
