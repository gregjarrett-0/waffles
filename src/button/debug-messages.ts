// Constants to be used as content for the associated component's `console.error` or `console.warn`.

export const MESSAGES = {
  EMPTY_CHILDREN: 'Button cannot have an empty string for children.',
  MISSING_ARIA_LABEL:
    'Button requires an `aria-label` when `isLoading={true}`.',
};
