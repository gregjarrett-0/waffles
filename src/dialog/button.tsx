import { forwardRef } from 'react';

import ButtonInternal, { ButtonProps } from './button-internal';

// TODO: Move contents to button-internal.tsx during modal refactor - ixTec

type ButtonComponent = <T extends React.ElementType = 'button'>(
  props: ButtonProps<T>,
) => JSX.Element | null;

const Button: ButtonComponent = forwardRef(ButtonInternal);

export default Button;
