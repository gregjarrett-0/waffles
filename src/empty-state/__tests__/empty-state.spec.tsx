import { render } from '@testing-library/react';

import { EmptyState } from '../index';
import { Paragraph } from '../../paragraph';
import { Group, Matrix, Branch } from '../../icon';
import { Button } from '../../button';
import { Rocket3d } from '../../asset';

jest.mock('../../icon', () => {
  return {
    Group: () => 'Group',
    Branch: () => 'Branch',
    Matrix: () => 'Matrix',
  };
});

jest.mock('../../asset', () => {
  return {
    Rocket3d: () => 'Rocket3d',
  };
});

describe('Dialog', () => {
  it('renders snapshot', async () => {
    const { container } = render(
      <EmptyState title="Title heading" image={<Rocket3d />}>
        <Paragraph>Lorem ipsum dolor sit amet.</Paragraph>
        <EmptyState.List>
          <EmptyState.ListItem icon={<Group />}>
            Auctor eu augue
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Branch />}>
            Tristique senectus et netus
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Matrix />}>
            Mus mauris vitae ultricies
          </EmptyState.ListItem>
        </EmptyState.List>
        <Button>Button</Button>
      </EmptyState>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders snapshot in vertical orientation', async () => {
    const { container } = render(
      <EmptyState
        title="Title heading"
        image={<Rocket3d />}
        orientation="vertical"
      >
        <Paragraph>Lorem ipsum dolor sit amet.</Paragraph>
        <EmptyState.List>
          <EmptyState.ListItem icon={<Group />}>
            Auctor eu augue
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Branch />}>
            Tristique senectus et netus
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Matrix />}>
            Mus mauris vitae ultricies
          </EmptyState.ListItem>
        </EmptyState.List>
        <Button>Button</Button>
      </EmptyState>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders snapshot in inverted', async () => {
    const { container } = render(
      <EmptyState title="Title heading" image={<Rocket3d />} inverted>
        <Paragraph inverted>Lorem ipsum dolor sit amet.</Paragraph>
        <EmptyState.List>
          <EmptyState.ListItem icon={<Group />}>
            Auctor eu augue
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Branch />}>
            Tristique senectus et netus
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Matrix />}>
            Mus mauris vitae ultricies
          </EmptyState.ListItem>
        </EmptyState.List>
        <Button>Button</Button>
      </EmptyState>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders snapshot in vertical orientation and inverted', async () => {
    const { container } = render(
      <EmptyState
        title="Title heading"
        image={<Rocket3d />}
        orientation="vertical"
        inverted
      >
        <Paragraph inverted>Lorem ipsum dolor sit amet.</Paragraph>
        <EmptyState.List>
          <EmptyState.ListItem icon={<Group />}>
            Auctor eu augue
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Branch />}>
            Tristique senectus et netus
          </EmptyState.ListItem>
          <EmptyState.ListItem icon={<Matrix />}>
            Mus mauris vitae ultricies
          </EmptyState.ListItem>
        </EmptyState.List>
        <Button>Button</Button>
      </EmptyState>,
    );

    expect(container).toMatchSnapshot();
  });
});
