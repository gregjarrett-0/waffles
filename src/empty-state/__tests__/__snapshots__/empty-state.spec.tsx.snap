// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog renders snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 4px;
  padding: 24px;
  gap: 16px;
  height: 100%;
  width: 100%;
  border: 1px dashed rgba(5, 25, 45, 0.1);
  color: #05192D;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 ul {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 72px;
  height: 72px;
}

.emotion-2 {
  max-width: 650px;
}

.emotion-3 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #05192D;
  font-weight: 800;
  font-size: 18px;
  margin-bottom: 8px;
  line-height: 1.25;
  line-height: 1.5;
  margin: 0;
  padding-bottom: 8px;
}

.emotion-4 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  line-height: 1.5;
  margin-bottom: 8px;
  color: #05192D;
  font-size: 14px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 16px 0 24px;
  gap: 8px;
}

.emotion-6 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  margin-left: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  gap: 16px;
  color: inherit;
  line-height: 1.25;
}

.emotion-6 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-9 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  color: #05192D;
  background-color: #03EF62;
  border-color: transparent;
}

.emotion-9::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-9:hover:not(:disabled) {
  background-color: #00D649;
}

.emotion-9:active:not(:disabled) {
  background-color: #03EF62;
}

.emotion-9:disabled {
  opacity: 0.6;
  cursor: default;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      Rocket3d
    </div>
    <div
      class="emotion-2"
    >
      <h2
        class="emotion-3"
      >
         Title Heading
      </h2>
      <p
        class="emotion-4"
      >
        Lorem ipsum dolor sit amet.
      </p>
      <ul
        class="emotion-5"
      >
        <li
          class="emotion-6"
        >
          Group
          Auctor eu augue
        </li>
        <li
          class="emotion-6"
        >
          Branch
          Tristique senectus et netus
        </li>
        <li
          class="emotion-6"
        >
          Matrix
          Mus mauris vitae ultricies
        </li>
      </ul>
      <button
        class="emotion-9"
        type="button"
      >
        <span
          class="emotion-10"
        >
           Button
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Dialog renders snapshot in inverted 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 4px;
  padding: 24px;
  gap: 16px;
  height: 100%;
  width: 100%;
  border: 1px dashed rgba(255, 255, 255, 0.1);
  color: #FFFFFF;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 ul {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 72px;
  height: 72px;
}

.emotion-2 {
  max-width: 650px;
}

.emotion-3 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  font-weight: 800;
  font-size: 18px;
  margin-bottom: 8px;
  line-height: 1.25;
  line-height: 1.5;
  margin: 0;
  padding-bottom: 8px;
}

.emotion-4 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  line-height: 1.5;
  margin-bottom: 8px;
  color: #FFFFFF;
  font-size: 14px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 16px 0 24px;
  gap: 8px;
}

.emotion-6 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  margin-left: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  gap: 16px;
  color: inherit;
  line-height: 1.25;
}

.emotion-6 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-9 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  color: #05192D;
  background-color: #03EF62;
  border-color: transparent;
}

.emotion-9::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-9:hover:not(:disabled) {
  background-color: #00D649;
}

.emotion-9:active:not(:disabled) {
  background-color: #03EF62;
}

.emotion-9:disabled {
  opacity: 0.6;
  cursor: default;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      Rocket3d
    </div>
    <div
      class="emotion-2"
    >
      <h2
        class="emotion-3"
      >
         Title Heading
      </h2>
      <p
        class="emotion-4"
      >
        Lorem ipsum dolor sit amet.
      </p>
      <ul
        class="emotion-5"
      >
        <li
          class="emotion-6"
        >
          Group
          Auctor eu augue
        </li>
        <li
          class="emotion-6"
        >
          Branch
          Tristique senectus et netus
        </li>
        <li
          class="emotion-6"
        >
          Matrix
          Mus mauris vitae ultricies
        </li>
      </ul>
      <button
        class="emotion-9"
        type="button"
      >
        <span
          class="emotion-10"
        >
           Button
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Dialog renders snapshot in vertical orientation 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 4px;
  padding: 24px;
  gap: 16px;
  height: 100%;
  width: 100%;
  border: 1px dashed rgba(5, 25, 45, 0.1);
  color: #05192D;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 ul {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 72px;
  height: 72px;
}

.emotion-2 {
  max-width: 650px;
}

.emotion-3 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #05192D;
  font-weight: 800;
  font-size: 18px;
  margin-bottom: 8px;
  line-height: 1.25;
  line-height: 1.5;
  margin: 0;
  padding-bottom: 8px;
}

.emotion-4 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  line-height: 1.5;
  margin-bottom: 8px;
  color: #05192D;
  font-size: 14px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 16px 0 24px;
  gap: 8px;
}

.emotion-6 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  margin-left: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  gap: 16px;
  color: inherit;
  line-height: 1.25;
}

.emotion-6 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-9 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  color: #05192D;
  background-color: #03EF62;
  border-color: transparent;
}

.emotion-9::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-9:hover:not(:disabled) {
  background-color: #00D649;
}

.emotion-9:active:not(:disabled) {
  background-color: #03EF62;
}

.emotion-9:disabled {
  opacity: 0.6;
  cursor: default;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      Rocket3d
    </div>
    <div
      class="emotion-2"
    >
      <h2
        class="emotion-3"
      >
         Title Heading
      </h2>
      <p
        class="emotion-4"
      >
        Lorem ipsum dolor sit amet.
      </p>
      <ul
        class="emotion-5"
      >
        <li
          class="emotion-6"
        >
          Group
          Auctor eu augue
        </li>
        <li
          class="emotion-6"
        >
          Branch
          Tristique senectus et netus
        </li>
        <li
          class="emotion-6"
        >
          Matrix
          Mus mauris vitae ultricies
        </li>
      </ul>
      <button
        class="emotion-9"
        type="button"
      >
        <span
          class="emotion-10"
        >
           Button
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Dialog renders snapshot in vertical orientation and inverted 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 4px;
  padding: 24px;
  gap: 16px;
  height: 100%;
  width: 100%;
  border: 1px dashed rgba(255, 255, 255, 0.1);
  color: #FFFFFF;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: left;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-0 ul {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 72px;
  height: 72px;
}

.emotion-2 {
  max-width: 650px;
}

.emotion-3 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #FFFFFF;
  font-weight: 800;
  font-size: 18px;
  margin-bottom: 8px;
  line-height: 1.25;
  line-height: 1.5;
  margin: 0;
  padding-bottom: 8px;
}

.emotion-4 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  line-height: 1.5;
  margin-bottom: 8px;
  color: #FFFFFF;
  font-size: 14px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 16px 0 24px;
  gap: 8px;
}

.emotion-6 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  margin-left: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  gap: 16px;
  color: inherit;
  line-height: 1.25;
}

.emotion-6 svg {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-9 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  vertical-align: middle;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-weight: 800;
  line-height: 1;
  margin: 0;
  padding: 0;
  outline: 0;
  border-style: solid;
  border-radius: 4px;
  border-width: 2px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: background-color 125ms ease-out;
  transition: background-color 125ms ease-out;
  cursor: pointer;
  height: 36px;
  min-width: 36px;
  width: auto;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 14px;
  color: #05192D;
  background-color: #03EF62;
  border-color: transparent;
}

.emotion-9::after {
  content: '';
  display: block;
  position: absolute;
  z-index: 1;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: -4px;
  border-radius: 6px;
  pointer-events: none;
  -webkit-transition: box-shadow 125ms ease-out;
  transition: box-shadow 125ms ease-out;
}

.emotion-9:hover:not(:disabled) {
  background-color: #00D649;
}

.emotion-9:active:not(:disabled) {
  background-color: #03EF62;
}

.emotion-9:disabled {
  opacity: 0.6;
  cursor: default;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      Rocket3d
    </div>
    <div
      class="emotion-2"
    >
      <h2
        class="emotion-3"
      >
         Title Heading
      </h2>
      <p
        class="emotion-4"
      >
        Lorem ipsum dolor sit amet.
      </p>
      <ul
        class="emotion-5"
      >
        <li
          class="emotion-6"
        >
          Group
          Auctor eu augue
        </li>
        <li
          class="emotion-6"
        >
          Branch
          Tristique senectus et netus
        </li>
        <li
          class="emotion-6"
        >
          Matrix
          Mus mauris vitae ultricies
        </li>
      </ul>
      <button
        class="emotion-9"
        type="button"
      >
        <span
          class="emotion-10"
        >
           Button
        </span>
      </button>
    </div>
  </div>
</div>
`;
